[objc] Issues compiling for arrays

A method with a signature like

`Boolean TryParseExact(System.String, System.String[], System.IFormatProvider, System.TimeSpan&)`

(from `TimeSpan`) generates code that won't compile

```
Debug/bindings.h:7819:58: error: expected ':'
+ (instancetype)parseExactString:(NSString *)input string[]:(NSArray <NSString *> *)formats iformatprovider:(id<System_IFormatProvider>)formatProvider;
```

The signature includes `string[]` which is suspicious (missing convert?)

This is blocking a fix for issue #176 since it's not possible to compile UnitsNet anymore

--- a/objcgen/NameGenerator.cs
+++ b/objcgen/NameGenerator.cs
@@ -181,6 +181,17 @@ namespace ObjC {
 			}
 		}
 
+		public static string GetParameterTypeName (Type t)
+		{
+			var name = t.Name;
+			if (t.IsArray)
+				return t.GetElementType ().Name + "Array";
+			else if (t.IsByRef)
+				return t.GetElementType ().Name + "Ref";
+			else
+				return t.Name;
+		}
+
 		public static string GetExtendedParameterName (ParameterInfo p, ParameterInfo [] parameters)
 		{
 			string pName = p.Name;

--- a/objcgen/objcgenerator-helpers.cs
+++ b/objcgen/objcgenerator-helpers.cs
@@ -37,7 +37,7 @@ namespace ObjC {
 					mono.Append (',');
 				}
 
-				string paramName = useTypeNames ? p.ParameterType.Name : p.Name;
+				string paramName = useTypeNames ? NameGenerator.GetParameterTypeName (p.ParameterType) : p.Name;
 				if ((method != null) && (n > 0 || !isExtension)) {
 					if (n == 0) {
 						if (useTypeNames || method.IsConstructor || (!method.IsSpecialName && !isOperator))

--- a/tests/managed/overloads.cs
+++ b/tests/managed/overloads.cs
@@ -315,4 +315,18 @@ namespace Overloads {
 			return Value.GetHashCode ();
 		}
 	}
+
+	public class ConflictingNamingOverload {
+
+		// objc: no need to call it (from xcode), the fact it compiles is fine
+		public static string Format (string [] values)
+		{
+			return String.Join (" ", values);
+		}
+
+		public static string Format (int [] values)
+		{
+			return String.Join (" ", values);
+		}
+	}
 }

